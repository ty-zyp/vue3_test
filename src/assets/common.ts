export function sum(a:number, b:number):number {
  return a + b
}

// compose 函数
export function compose(...funcs:Array<Function>){
    return function anonymous(val:any){
    if(funcs.length===0) return val;
    if(funcs.length===1) return funcs[0](val);
    return funcs.reverse().reduce((N,item)=>{
      return item(N);
    },val)
  }
}
interface User{
  userName: string,
  name: string,
  project: string[],
  coolness?: number,
  favoriteFood?:string
}


const findUser = (userName:string): User | undefined =>{
  const users:User[] = [{
      userName: 'zs',
      name: 'zhangsan',
      project: ['a', 'b'],
      favoriteFood:'nodes'
    },
    {
      userName: 'ls',
      name: 'lisi',
      project: ['c', 'd'],
      favoriteFood:'mifan'
    }
  ]
  return users.find(user => {
    return user.userName === userName;
  })
}

export const loadUserData = async (userName:string) => {
  const user = await findUser(userName);
  if (!user) {
    throw new Error('no user found');
  }
  user.coolness = userName === 'ls' ? 100 : -1;
  return user;
}