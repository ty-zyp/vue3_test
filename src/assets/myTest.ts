import { test } from 'vitest'

const todos = []
const archive = []

export const myTest = test.extend({
  // eslint-disable-next-line no-empty-pattern
  todos: async ({}, use) => {
    // 在每次测试函数运行之前设置固定装置
    todos.push(1, 2, 3)

    // 使用固定装置的值
    await use(todos)

    // 在每次测试函数运行之后清除固定装置
    todos.length = 0
  },
  archive,
})

// 在此处可以扩展
export const myTest2 = myTest.extend({
  settings: {
    // ...
  },
})