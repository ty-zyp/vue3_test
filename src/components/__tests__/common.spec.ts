import { describe, it, expect,test } from 'vitest'
// import { mount } from '@vue/test-utils'
import { sum,compose, loadUserData} from '../../assets/common' // src\components\HelloWorld.vue

// 写法一：
// test('adds 1 + 2 to equal 3', () => {
//   expect(sum(1,2)).toBe(3)
// })

// const add1 = (x:number) => x + 1;
// const mul3 = (x:number) => x * 3;
// const div2 = (x:number) => x / 2;
// test('compose函数 3 equal 5.5', () => {
//   expect(compose(add1,mul3,div2)(3)).toBe(5.5)
// })

// 写法二：在测试中给it使用 concurrent 来并行运行
// describe('common.ts', () => {
//   test.concurrent('adds 1 + 2 to equal 3', () => {
//     expect(sum(1, 2)).toBe(3)
//   });
//   const add1 = (x:number) => x + 1;
//   const mul3 = (x:number) => x * 3;
//   const div2 = (x:number) => x / 2;
//   test.concurrent('compose函数 3 equal 5.5', () => {
//     expect(compose(add1, mul3, div2)(3)).toBe(5.5)
//   });
// })

// 写法三：在测试中给 describe 使用 concurrent 来并行运行
describe.concurrent('common.ts', () => {
  test('adds 1 + 2 to equal 3', () => {
    expect(sum(1, 2)).toBe(3)
  });
  const add1 = (x:number) => x + 1;
  const mul3 = (x:number) => x * 3;
  const div2 = (x:number) => x / 2;
  test('compose函数 3 equal 5.5', () => {
    expect(compose(add1, mul3, div2)(3)).toBe(5.5)
  });
})

describe('api-test', () => {
  test('获取接口数据', async () => {
    const result = await loadUserData('ls');
    expect(result).toEqual({
      userName: 'ls',
      name: 'lisi',
      project: ['c', 'd'],
      coolness: 100,
      favoriteFood: 'mifan'
    })
  })
  test('common 2', async () => {
    const zs = await loadUserData('zs');
    const ls = await loadUserData('ls');
    expect(zs.coolness).toBe(-1);
    expect(ls.coolness).toBe(100);
  })
  // test.fails('common 3', async () => {
  test('common 3', async () => {
    expect(async () => {
      await loadUserData('ss')
    }).rejects.toThrowError();
  })
  test('common 4', async () => {
    expect(async () => {
      await loadUserData('ss')
    }).rejects.toThrowError('no user found'); // 括号内容只要存在于抛出的内容上及测试通过，如 user 在 no user found 中能找到
  })
  test('common快照-1', async () => {
    const result = await loadUserData('ls');
    expect(result).toMatchSnapshot()
  })
})