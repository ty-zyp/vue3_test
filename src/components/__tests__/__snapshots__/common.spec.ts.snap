// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`api-test > common快照-1 1`] = `
{
  "coolness": 100,
  "favoriteFood": "mifan",
  "name": "lisi",
  "project": [
    "c",
    "d",
  ],
  "userName": "ls",
}
`;
